client/src/app/pages/questionnaires/questionnaires.tsx

```diff
@@ -0,0 +1,459 @@
+                            what: t("terms.YAMLTemplate"),
+                          })}
+                        </Button>
+                      </ToolbarItem>
+                    ) : null
+                  }
+                </ToolbarGroup>
+                <ToolbarItem {...paginationToolbarItemProps}>
+                  <SimplePagination
+                    idPrefix="questionnaire-table"
+                    isTop
+                    paginationProps={paginationProps}
+                  />
+                </ToolbarItem>
+              </ToolbarContent>
+            </Toolbar>
+            <Table {...tableProps} aria-label="Assement table">
+              <Thead>
+                <Tr>
+                  <TableHeaderContentWithControls {...tableControls}>
+                    <Th {...getThProps({ columnKey: "required" })} />
+                    <Th {...getThProps({ columnKey: "name" })} />
+                    <Th {...getThProps({ columnKey: "questions" })} />
+                    <Th {...getThProps({ columnKey: "rating" })} />
+                    <Th {...getThProps({ columnKey: "dateImported" })} />
+                  </TableHeaderContentWithControls>
+                </Tr>
+              </Thead>
+              <ConditionalTableBody
+                isLoading={isFetching}
+                isError={!!fetchError}
+                isNoData={currentPageItems.length === 0}
+                noDataEmptyState={
+                  <EmptyState variant="sm">
+                    <EmptyStateIcon icon={CubesIcon} />
+                    <Title headingLevel="h2" size="lg">
+                      No questionnaire available
+                    </Title>
+                    <EmptyStateBody>
+                      Use the import button above to add your questionnaire.
+                    </EmptyStateBody>
+                  </EmptyState>
+                }
+                numRenderedColumns={numRenderedColumns}
+              >
+                {currentPageItems?.map((questionnaire, rowIndex) => {
+                  return (
+                    <Tbody key={questionnaire.id}>
+                      <Tr>
+                        <TableRowContentWithControls
+                          {...tableControls}
+                          item={questionnaire}
+                          rowIndex={rowIndex}
+                        >
+                          <Td
+                            width={10}
+                            {...getTdProps({ columnKey: "required" })}
+                          >
+                            {questionnaire.required}
+                            <Switch
+                              id={`required-switch-${rowIndex.toString()}`}
+                              label="Yes"
+                              labelOff="No"
+                              isChecked={questionnaire.required}
+                              onChange={(_event, _checked) => {
+                                updateQuestionnaire(
```

Comment:
Are we hitting the api on every change? I think currently we dont submit the questionnaire until the end of the assessment.

Reply:
This is not assessment it's questionnaire page, although the menu is dubbed Assessment.
So when the user is switching the `Required` field we need to update the backend.
I'm not sure yet how the final Questionnaire data is going to look like, for now I assume it's part of the Questionnaire data.

Reply:
Oh ok makes sense.